### PR TITLE
Add docs for ProductData properties

### PR DESCRIPTION
**Pull request by dstaley**
## "Ready for Review" checklist

- [ ] The Vercel preview link has been added to this PR's description
- [ ] The Asana task has been added to this PR's description
- [ ] This PR's link has been added to the "📐 Pull Request" field on the Asana task
- [ ] This Vercel preview link has been added to the "🔍 Preview" field on the Asana task
- [ ] The description template has been filled in below

---

## Relevant links

- [Preview link](https://dev-portal-git-BRANCH_NAME-hashicorp.vercel.app/PATH_TO_VIEW) 🔎
- [Asana task](url) 🎟️

## What

This PR adds JSDoc comments for all the properties used in `src/data/<product>.json` files, and updates the README within that folder to reference the types.

### CODE REVIEW

**Comment by dstaley** [comment]
@ashleemboyer Feel free to ignore my guesses at the DevDot-specific properties 😅 Happy to update those to match what you come up with.

File: src/data/types.d.ts [hunk 1]
```diff
@@ -8,65 +11,318 @@ export type ProductSlug =
...
+   * @see {@link Sidebar}
+   */
+  sidebar: Sidebar
+
+  /**
+   * An array of items describing the primary top-level navigation for this
+   * product in Dev Portal.
+   * @see {@link NavigationHeaderItem}
+   */
+  navigationHeaderItems: NavigationHeaderItem[]
```
Comment (ashleemboyer) [inline src/data/types.d.ts:87]: These are defined in another file and documenting them is part of another task I'm working on. They can be removed!

File: src/data/types.d.ts [hunk 2]
```diff
@@ -8,65 +11,318 @@ export type ProductSlug =
...
+
+  /**
+   * The relative path for this navigation item.
+   */
+  path: string
+
+  /**
+   * A unique identifier used as a key for React.
+   */
+  id: string
```
Comment (ashleemboyer) [inline src/data/types.d.ts:228]: Same as previous comment, this is already defined elsewhere and I'll handle documenting it in my other task. 😊

File: src/data/types.d.ts [hunk 3]
```diff
@@ -8,65 +11,318 @@ export type ProductSlug =
...
+   */
+  landingPageNavData: LandingPageNavDatum[]
+
+  /**
+   * An array of links to render under the Resources section in the sidebar on
+   * this product's Dev Portal landing page.
+   * @see {@link ResourcesNavDatum}
+   */
+  resourcesNavData: ResourcesNavDatum[]
+}
```
Comment (ashleemboyer) [inline src/data/types.d.ts:262]: Same as previous. 😊

**Review by ashleemboyer — CHANGES_REQUESTED** [review]

**Review by ashleemboyer — APPROVED** [review]
🥳

**Pull request merged**